game-core/src/test/java/games/strategy/util/LocalizeHtmlTest.java

```diff
@@ -0,0 +1,78 @@
+package games.strategy.util;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.mockito.Mockito.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.net.URL;
+
+import org.junit.jupiter.api.Test;
+
+import games.strategy.triplea.ResourceLoader;
+
+class LocalizeHtmlTest {
```

Comment:
It looks like these tests were added after-the-fact.  Did you happen to run them against the original implementation to ensure the new implementation has the same behavior?

Reply:
I almost expected you to ask this question 😅
You are correct, I added the tests afterwards, and no I didn't run them against the original implementation.
Because the new implementation  should "just work" and keeping the exact same behaviour wasn't a necessity IMO, I didn't even bother.
The original version for example required xhtml i.e. `<img />` with the closing `/` while the new implementation works without it.
But this reminds me of something: I'd assume swing isn't very strict when it comes to html parsing, so I'd assume everything to work even when not closing the tag at all. So we could explicitly allow that as well, and add tests to enforce whichever behaviour we prefer.